src/App.tsx

```diff
@@ -15,37 +15,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const showList = (arr: string[]) =>
+  arr.map(good => (
+    <li key={good} data-cy="Good">
+      {good}
+    </li>
+  ));
+
+const defaultList = showList(goodsFromServer);
+
+export const App = () => {
+  const [actualList, setActualList] = useState(defaultList);
+  const [actualListArr, setActualListArr] = useState([...goodsFromServer]);
+  const [sortMethod, setSortMethod] = useState<string | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortArray = (arr: string[], method: string): string[] => {
+    let sortedArr;
+
+    switch (method) {
+      case 'alphabetically':
+        sortedArr = [...arr].sort((a, b) => a.localeCompare(b));
+        break;
+      case 'byLength':
+        sortedArr = [...arr].sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedArr = arr;
+    }
+
+    return sortedArr;
+  };
+
+  const handleSort = (method: string) => {
+    let sortedArr = sortArray(goodsFromServer, method);
+
+    if (isReversed) {
+      sortedArr = [...sortedArr].reverse();
+    }
+
+    setSortMethod(method);
+    setActualListArr(sortedArr);
+    setActualList(showList(sortedArr));
+  };
+
+  const handleReverse = () => {
+    const reversedArr = [...actualListArr].reverse();
+
+    setIsReversed(!isReversed);
+    setActualListArr(reversedArr);
+    setActualList(showList(reversedArr));
+  };
+
+  const handleReset = () => {
+    setSortMethod(null);
+    setIsReversed(false);
+    setActualListArr([...goodsFromServer]);
+    setActualList(defaultList);
+  };
+
+  const getButtonClass = (method: string) => {
+    if (sortMethod === method && method === 'alphabetically') {
+      return 'button is-info';
+    }
+
+    if (method === 'alphabetically') {
+      return 'button is-info is-light';
+    }
+
+    if (sortMethod === method && method === 'byLength') {
+      return 'button is-success';
+    }
+
+    if (method === 'byLength') {
+      return 'button is-success is-light';
+    }
+
+    return 'button is-light';
```

Comment:
The function 'getButtonClass' should start with a verb as per the [CODE STYLE] checklist. A name like 'determineButtonClass' would be more descriptive of its functionality.